File: android/app/src/main/java/org/bosf/pooch/data/local/entities/Dog.kt
package org.bosf.pooch.data.local.entities

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters

@Entity(tableName = "dogs")
@TypeConverters(Converters::class)
data class Dog(
    @PrimaryKey val id: String,
    val name: String,
    val breed: String?,
    val age: Int?,
    val weight: Double?,
    val activityLevel: String?,
    val allergies: List<String>,
    val healthConditions: List<String>,
    val photoUrl: String?,
    val createdAt: String,
    val updatedAt: String
)

File: ios/PoochScan/PoochScan/Models/Dog.swift
import Foundation

struct Dog: Codable, Identifiable {
    let id: String
    var name: String
    var breed: String?
    var age: Int?
    var weight: Double?
    var activityLevel: String?
    var allergies: [String]
    var healthConditions: [String]
    var photoUrl: String?
    let userId: String
    let createdAt: String
    
    enum CodingKeys: String, CodingKey {
        case id, name, breed, age, weight
        case activityLevel = "activity_level"
        case allergies
        case healthConditions = "health_conditions"
        case photoUrl = "photo_url"
        case userId = "user_id"
        case createdAt = "created_at"
    }
    
    init(
        id: String = "",
        name: String,
        breed: String? = nil,
        age: Int? = nil,
        weight: Double? = nil,
        activityLevel: String? = nil,
        allergies: [String] = [],
        healthConditions: [String] = [],
        photoUrl: String? = nil,
        userId: String = "",
        createdAt: String = ""
    ) {
            self.id = id
            self.name = name
            self.breed = breed
            self.age = age
            self.weight = weight
            self.activityLevel = activityLevel
            self.allergies = allergies
            self.healthConditions = healthConditions
            self.photoUrl = photoUrl
            self.userId = userId
            self.createdAt = createdAt
        }
}
